### PR TITLE
Support old versions of curl and better hostname support

### PR DESCRIPTION
**Pull request by mcnamamj**
@francoisl @tonyzakula 

Whattdya think?

### CODE REVIEW

**Comment by francoisl** [comment]
Yep. In this case we could also directly provide them with a url-encoded version of the template maybe, to make their life easier?

**Comment by francoisl** [comment]
Actually, just found this to urlencode in bash http://stackoverflow.com/a/10660730

``` bash
rawurlencode() {
  local string="${1}"
  local strlen=${#string}
  local encoded=""

  for (( pos=0 ; pos<strlen ; pos++ )); do
     c=${string:$pos:1}
     case "$c" in
        [-_.~a-zA-Z0-9] ) o="${c}" ;;
        * )               printf -v o '%%%02x' "'$c"
     esac
     encoded+="${o}"
  done
  echo "${encoded}"    # You can either set a return variable (FASTER) 
  REPLY="${encoded}"   #+or echo the result (EASIER)... or both... :p
}
```

We could maybe adapt it and put it in the script and do it ourselves, so that they don't have to worry about passing a url-encoded file.

**Comment by tonyzakula** [comment]
all of it looks great to me

**Comment by mcnamamj** [comment]
Cool find @francoisl !  I'll integrate it in.

**Comment by mcnamamj** [comment]
Review again, updated to automatically switch to bash urlencode if the curl version is bad or to manually do it if there is a command line argument passed in.

/cc: @righdforsa @jessecollier I'm awful at bash and detest the language, but can you fact check what I did here?

File: expensify_export.sh [hunk 1]
```diff
@@ -172,7 +193,13 @@ function set_defaults() {
     EXPORT_FILE="expensify_export.$(date +'%Y%m%d').csv"
 
     # hostname for the expensify integrations server
-    HOSTNAME="integrations.expensify.com"
+    HOSTNAME="https://integrations.expensify.com"
```
Comment (mcnamamj) [inline expensify_export.sh:230]: switching this allows me to have a debug mode on my local env where HOSTNAME="http://integrations.expensify.com:8080" (switch protocol and port).  Very handy.

**Comment by francoisl** [comment]
Cool, seems good. Just one more thing, can we add some information about the version of the script (at the top, as a comment maybe)? That could be useful for companies like Carmanah, when we give them an updated version of the script.

**Comment by mcnamamj** [comment]
Jesse founda  logic bug in the min version checking for curl.  Need to hold off for a minute.

Will add a "--version" flag.

**Comment by jessecollier** [comment]
https://gist.github.com/jessecollier/5880449

``` bash
version_cmp () {
    if [[ $1 == $2 ]]
    then
        return 0
    fi
    local IFS=.
    local i ver1=($1) ver2=($2)
    # fill empty fields in ver1 with zeros
    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
    do
        ver1[i]=0
    done
    for ((i=0; i<${#ver1[@]}; i++))
    do
        if [[ -z ${ver2[i]} ]]
        then
            # fill empty fields in ver2 with zeros
            ver2[i]=0
        fi
        if ((10#${ver1[i]} > 10#${ver2[i]}))
        then
            return 1
        fi
        if ((10#${ver1[i]} < 10#${ver2[i]}))
        then
            return 2
        fi
        if ((10#${ver1[i]} >= 10#${ver2[i]}))
        then
            return 3
        fi
        if ((10#${ver1[i]} <= 10#${ver2[i]}))
        then
            return 3
        fi
    done
    return 0
}
```

**Comment by mcnamamj** [comment]
Updated to use better version of version comparison.  Also added the --version flag.  Please review again.

**Comment by francoisl** [comment]
Should `--version` be included in `usage()`?

**Pull request merged**